### PR TITLE
Refactor: don't prop-drill shouldCollectX

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Refactor to remove prop-drilling of `shouldCollectEmail` and `shouldCollectName`

## Justify why they are needed

Smaller API, less code

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4180?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4180** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4180?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4178** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4178?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/CheckoutPage/CheckoutPage.tsx [hunk 1]
```diff
@@ -140,21 +141,13 @@ const CheckoutPage = (props: CheckoutPageProps) => {
...
 // Prop-drilling so many props is awkward, consider refactoring to group them somehow
-const CheckoutForm = ({
-  checkoutSteps,
-  onSignError,
-  shopSession,
-  ssn,
-  shouldCollectEmail,
-  shouldCollectName,
-}: CheckoutFormProps) => {
+const CheckoutForm = ({ checkoutSteps, onSignError, shopSession, ssn }: CheckoutFormProps) => {
```
Comment (Youakeem) [inline apps/store/src/components/CheckoutPage/CheckoutPage.tsx:150]: If we have `shopSession.custome` can we also get the SSN off of it?

Comment (Youakeem) [reply apps/store/src/components/CheckoutPage/CheckoutPage.tsx:150]: Related to that, can we use `useShopSession` to access it directly?

Comment (alebedev) [reply apps/store/src/components/CheckoutPage/CheckoutPage.tsx:150]: `useShopSession` is async, so we'd need to handle a case where session is not ready - we don't want it during checkout

Comment (Youakeem) [reply apps/store/src/components/CheckoutPage/CheckoutPage.tsx:150]: Got it!

**Comment by alebedev** [comment]
Logically following up on "don't prop drill at checkout" makes a large and messy change. Don't have time for it right now and not sure it's worth the effort until we have other reasons to refactor

**Pull request closed without merging**